sphinx/domains/std/__init__.py

```diff
@@ -996,6 +996,35 @@ def resolve_any_xref(self, env: BuildEnvironment, fromdocname: str,
                                                    labelid, contnode)))
         return results
 
+    def _intersphinx_adjust_object_types(self, objtypes: list[str]) -> None:
+        # we adjust the object types for backwards compatibility
+        if 'cmdoption' in objtypes:
+            # until Sphinx-1.6, cmdoptions are stored as std:option
+            objtypes.append('option')
+
+    def _intersphinx_resolve_xref_lookup(self, store: dict[str, dict[str, Any]],
+                                         target: str, objtypes: list[str]
+                                         ) -> Any | None:
+        # Semi-haxy overwriting of the private method in Domain, as we only need to do
```

Comment:
You could say that the method is actually protected :)

Reply:
Indeed, I have added some internal documentation in the base class to make it clear that it should be changed carefully.